Grails 7.0.0-SNAPSHOT PublishToMavenLocal Fails

### Issue description

Execution failed for task ':grails-dependencies:installToHomeDist'. on Grails 7.0.0-SNAPSHOT w/ Gradle 8.10

```
        tasks.register("installToHomeDist", Upload) {
            configuration = configurations.archives
            repositories {
                flatDir name: 'libs', dirs: distInstallDir
            }
        }

        tasks.withType(PublishToMavenLocal).configureEach {
            dependsOn installToHomeDist
            doLast {
                ant.copy(todir: homeDistDir, flatten: true, includeEmptyDirs: false) {
                    fileset dir: distInstallDir
                }
            }
        }
```

11:02:27 AM: Executing 'publishToMavenLocal'...

> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:compileGroovy UP-TO-DATE
> Task :buildSrc:processResources NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
> Task :grails-web-mvc:processResources NO-SOURCE
> Task :grails-encoder:processResources NO-SOURCE
> Task :grails-web-databinding:processResources UP-TO-DATE
> Task :grails-console:processResources NO-SOURCE
> Task :grails-docs:compileJava NO-SOURCE
> Task :grails-databinding:compileJava NO-SOURCE
> Task :grails-gradle-model:compileJava NO-SOURCE
> Task :grails-logging:processResources NO-SOURCE
> Task :grails-web:processResources UP-TO-DATE
> Task :grails-codecs:compileJava NO-SOURCE
> Task :grails-plugin-codecs:processResources UP-TO-DATE
> Task :grails-web-url-mappings:processResources UP-TO-DATE
> Task :grails-plugin-controllers:processResources UP-TO-DATE
> Task :grails-web-common:processResources UP-TO-DATE
> Task :grails-plugin-mimetypes:processResources UP-TO-DATE
> Task :grails-plugin-databinding:processResources NO-SOURCE
> Task :grails-bootstrap:compileJava NO-SOURCE
> Task :grails-plugin-datasource:processResources UP-TO-DATE
> Task :grails-plugin-i18n:processResources UP-TO-DATE
> Task :grails-plugin-validation:processResources UP-TO-DATE
> Task :grails-spring:processResources NO-SOURCE
> Task :grails-plugin-url-mappings:processResources UP-TO-DATE
> Task :grails-plugin-interceptors:processResources NO-SOURCE
> Task :grails-core:writeGrailsProperties FROM-CACHE
> Task :grails-plugin-rest:processResources UP-TO-DATE
> Task :grails-plugin-domain-class:processResources UP-TO-DATE
> Task :grails-dependencies:generatePomFileForMavenPublication
> Task :grails-dependencies:installToHomeDist FAILED
> Task :grails-bom:generatePomFileForMavenPublication
> Task :grails-bom:installToHomeDist FAILED
> Task :grails-logging:sourcesJar
> Task :grails-encoder:sourcesJar
> Task :grails-plugin-rest:sourcesJar
> Task :grails-plugin-i18n:sourcesJar
> Task :grails-plugin-codecs:sourcesJar
> Task :grails-bootstrap:compileGroovy UP-TO-DATE
> Task :grails-console:sourcesJar
> Task :grails-core:processResources
> Task :grails-plugin-domain-class:sourcesJar
> Task :grails-plugin-databinding:sourcesJar
> Task :grails-docs:compileGroovy UP-TO-DATE
> Task :grails-codecs:compileGroovy UP-TO-DATE
> Task :grails-databinding:compileGroovy UP-TO-DATE
> Task :grails-gradle-model:compileGroovy UP-TO-DATE
> Task :grails-plugin-interceptors:sourcesJar
> Task :grails-plugin-datasource:sourcesJar
> Task :grails-plugin-mimetypes:sourcesJar
> Task :grails-plugin-controllers:sourcesJar

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.10/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
38 actionable tasks: 17 executed, 1 from cache, 20 up-to-date

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':grails-bom:installToHomeDist'.
> The legacy `Upload` task was removed in Gradle 8. Please use the `maven-publish` or `ivy-publish` plugin instead. For more on publishing on maven repositories, please refer to https://docs.gradle.org/8.10/userguide/publishing_maven.html#publishing_maven in the Gradle documentation.
  For more on publishing on ivy repositories, please refer to https://docs.gradle.org/8.10/userguide/publishing_ivy.html#publishing_ivy in the Gradle documentation.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':grails-dependencies:installToHomeDist'.
> The legacy `Upload` task was removed in Gradle 8. Please use the `maven-publish` or `ivy-publish` plugin instead. For more on publishing on maven repositories, please refer to https://docs.gradle.org/8.10/userguide/publishing_maven.html#publishing_maven in the Gradle documentation.
  For more on publishing on ivy repositories, please refer to https://docs.gradle.org/8.10/userguide/publishing_ivy.html#publishing_ivy in the Gradle documentation.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org.
==============================================================================

BUILD FAILED in 1s
11:02:28 AM: Execution finished 'publishToMavenLocal'.

--- a/build.gradle
+++ b/build.gradle
@@ -417,15 +417,7 @@ subprojects { project ->
             shouldRunAfter(tasks.withType(Sign))
         }
 
-        tasks.register("installToHomeDist", Upload) {
-            configuration = configurations.archives
-            repositories {
-                flatDir name: 'libs', dirs: distInstallDir
-            }
-        }
-
         tasks.withType(PublishToMavenLocal).configureEach {
-            dependsOn installToHomeDist
             doLast {
                 ant.copy(todir: homeDistDir, flatten: true, includeEmptyDirs: false) {
                     fileset dir: distInstallDir

